reference/forms/types/options/empty_data.rst.inc

```diff
@@ -9,24 +9,35 @@ empty_data
 
 DEFAULT_PLACEHOLDER
 
-This option determines what value the field will return when the submitted
-value is empty.
-
-But you can customize this to your needs. For example, if you want the
-``gender`` choice field to be explicitly set to ``null`` when no value is
-selected, you can do it like this::
-
-    $builder->add('gender', 'choice', array(
-        'choices' => array(
-            'm' => 'Male',
-            'f' => 'Female'
-        ),
-        'required'    => false,
-        'placeholder' => 'Choose your gender',
-        'empty_data'  => null
+This option determines what value the field will *return* when the submitted
+value is empty (or missing). It does not set an initial value if none is
+provided when the form is rendered in a view.
+
+This means it helps you handling form submission with blank fields. For
+example, if you want the ``name`` field to be explicitly set to ``John Doe``
+when no value is selected, you can do it like this::
+
+    $builder->add('name', null, array(
+        'required'   => false,
+        'empty_data' => 'John Doe',
     ));
 
+This will still render an empty text box, but upon submission the ``John Doe``
+value will be set. Use the ``data`` or ``placeholder`` options to show this
+initial value in the rendered form.
+
+If a form is compound, you can set ``empty_data`` as an array with field names
```

Comment:
when `data_class` is null, otherwise it has to match the class set for this option.